Pear/Profile/GroupsSettingsViewController.swift

```diff
@@ -0,0 +1,259 @@
+//
+//  AddingGroupSettingsViewController.swift
+//  Pear
+//
+//  Created by Tiffany Pan on 10/18/22.
+//  Copyright © 2022 cuappdev. All rights reserved.
+//
+
+import UIKit
+import SnapKit
+
+class GroupsSettingsViewController: UIViewController {
+
+    // MARK: - Private Data vars
+    private weak var delegate: didUpdateGroupsDelegate?
+    private var displayedGroups: [Group] = []
+    private var groups: [Group] = []
+    private var selectedGroups: [Group] = []
+    private var updatedUser: UserV2
+    
+    // MARK: - Private View Vars
+    private let backButton = UIButton()
+    private let clubLabel = UILabel()
+    private let fadeTableView = FadeWrapperView(
+        UITableView(),
+        fadeColor: .backgroundLightGreen
+    )
+    private let saveButton = DynamicButton()
+    private let searchBar = UISearchBar()
+    private let subtitleLabel = UILabel()
+
+    init(updatingUser: UserV2, delegate: didUpdateGroupsDelegate) {
+        self.selectedGroups = updatingUser.groups
+        self.delegate = delegate
+        self.updatedUser = updatingUser
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
```

Comment:
When creating subviews instead of configuring all the subviews in `viewDidLoad` and setting up the constraints for each in a separate function, create a function for setting up each subview and doing the constraints in that function.

For example, you would have `setupSearchBar() {

// Search Bar Configuration and adding subview
// Search bar constraints setup

`

This keeps the searchBar constraint and setup code close together and easier to locate